### ISSUE TITLE
RPO Loss Inconsistency

### ISSUE THREAD

**Issue by chrisliu298** [huggingface/trl#1868]
In the [RPO paper](https://arxiv.org/abs/2404.19733), the loss is calculated as the sum of the vanilla DPO loss and the NLL loss on the chosen completion, as shown in the image below. Here, the NLL loss is weighted by alpha.

![2024-07-24 at 17 10 53](https://github.com/user-attachments/assets/67a692dc-9419-4a12-9ba6-f8e7b4bcfbe9)

In the implementation of [DPOTrainer](https://github.com/chrisliu298/trl/blob/ea0d3985a04ff9cfccde01d05121dfd04dcb3c14/trl/trainer/dpo_trainer.py#L1387-L1388), the loss is calculated as the DPO loss weighted by alpha, plus the NLL loss:

```python
if self.args.rpo_alpha is not None:
    losses = losses * self.args.rpo_alpha + policy_nll_loss
```

If I understand correctly, the above formulation is different from the loss function in the official paper. I wonder why the weight alpha is moved to the DPO loss in TRL.

**Comment by Weiyun1025** [comment]
In the v1 version of the RPO paper, the alpha is applied to the DPO Loss, whereas in their v3 paper, this hyperparameter is applied to the NLL Loss. This is quite confusing...

**Comment by qgallouedec** [comment]
Since most users will naturally come across the latest version of the paper, I think it would be appropriate for the TRL implementation to be consistent with it. WDYT @kashif?

**Comment by kashif** [comment]
yes  agree let me check
